OctopusDeploy/OctopusCLI
# Bug promote release - add latestSuccessful to promote-release command

Fixes - https://github.com/OctopusDeploy/OctopusCLI/issues/85

Currently the when promoting a release, the cli always pics the latest release regardless of whether that version has failed. We now add a `latestSuccessful` option to the command, thereby allowing you to pick the latest successful release for promotion. 

### How to test -

You will need this version of server - https://github.com/OctopusDeploy/OctopusDeploy/pull/6363

1) Create 2 - 3 environments
2) Create a failed deployment to 1st source environment, then success, and then a failed. So you end up with 3 releases, with 2 second release that is successful
3) promote the release without the `latestSuccessful` flag, this should fail
4) promote the release with the `latestSuccessful` flag, this should pick the second release that was successful
